htp/htp_transaction.c

```diff
@@ -834,8 +834,19 @@ htp_status_t htp_tx_res_process_body_data_ex(htp_tx_t *tx, const void *data, siz
             if (tx->connp->out_decompressor == NULL || tx->connp->out_decompressor->decompress == NULL)
                 return HTP_ERROR;
 
+            struct timeval before, after;
+            gettimeofday(&before, NULL);
             // Send data buffer to the decompressor.
             tx->connp->out_decompressor->decompress(tx->connp->out_decompressor, &d);
+            gettimeofday(&after, NULL);
+            //check for overflows ?
```

Comment:
I think you should be able to use `timercmp` here. See `man timercmp` or the existing examples in our code.

Reply:
Why is that ?
I could rather use `timersub` but I do not think it is better resilient against the overflow from the race condition (if you change the system time during this processing)

Reply:
Wasn't thinking about time changes, just about a cleaner way to deal with the `struct timeval` then the manual check.